### PR TITLE
pkg/types/path Distinguish between /topicA  and /topicA/

### PR DESCRIPTION
**Pull request by YujiOshima**
@chungers In #429, behavior seems to be different from what we were discussing.
It can not distinguish between topic /A and topic /A/.
Therefore, subscribing `/toppicA` will always subscribe all of `/topicA/` and below subtopic, you can not subscribe only `/topicA`.
I added test `TestBrokerSubscriberExactMatchTopic`. This is the test I want to do in this PR.

Signed-off-by: Yuji Oshima <yuji.oshima0x3fd@gmail.com>

### CODE REVIEW

File: pkg/broker/server/sse.go [hunk 1]
```diff
@@ -239,13 +250,17 @@ func (b *Broker) run() {
...
+					chset, ok := value.(map[chan []byte]clientValue)
 					if !ok {
-
 						panic("assert-failed")
 					}
 
-					for ch := range chset {
+					for ch, value := range chset {
+						if value.exactMatch && event.topic != key {
+							fmt.Printf("skip notifier event.topic%v,  key %v\n", event.topic, key)
```
Comment (chungers) [inline pkg/broker/server/sse.go]: Can you remove this?

File: pkg/broker/server/sse.go [hunk 2]
```diff
@@ -239,13 +250,17 @@ func (b *Broker) run() {
...
+							return false
+						}
+						fmt.Printf("notifier event.topic%v,  key %v\n", event.topic, key)
```
Comment (chungers) [inline pkg/broker/server/sse.go]: Same here.  Can you remove this line?

File: pkg/broker/server/sse.go [hunk 3]
```diff
@@ -80,6 +87,13 @@ func clean(topic string) string {
 	return topic
 }
 
+func checkExactMatch(topic string) bool {
```
Comment (chungers) [inline pkg/broker/server/sse.go]: This can be written as one line right?

File: pkg/broker/server/sse.go [hunk 4]
```diff
@@ -167,7 +181,7 @@ func (b *Broker) run() {
 			// Disconnect all clients
 			b.clients.Walk(
 				func(key string, value interface{}) bool {
-					chset, ok := value.(map[chan []byte]int)
+					chset, ok := value.(map[chan []byte]clientValue)
```
Comment (chungers) [inline pkg/broker/server/sse.go]: can't we just have a simpler value in the radix tree without the type `clientValue`?  

We really just need something like `map[chan[]byte]bool`, where the value bool is exactMatch or not.

File: pkg/broker/server/sse.go [hunk 5]
```diff
@@ -16,15 +17,21 @@ import (
...
+	ch         chan []byte
 }
 
 type event struct {
 	topic string
 	data  []byte
 }
 
+type clientValue struct {
+	value      int
```
Comment (chungers) [inline pkg/broker/server/sse.go]: We probably don't need this struct type at all.  We only need to store the value in the radix tree as `map[chan []byte]bool` where bool is just the `exactMatch` value.  We don't need this `value` field...

File: pkg/broker/server/sse.go (hunk 3 above)
Comment (YujiOshima) [inline pkg/broker/server/sse.go]: Sorry! I removed.

File: pkg/broker/server/sse.go (hunk 2 above)
Comment (YujiOshima) [inline pkg/broker/server/sse.go]: Sorry! I removed.

File: pkg/broker/server/sse.go (hunk 4 above)
Comment (YujiOshima) [inline pkg/broker/server/sse.go]: I wrote this because I was not sure how to use the original `int`.  Fixed, thanks!

**Pull request merged**